app/controllers/assignments_controller.rb

```diff
@@ -45,106 +46,43 @@ def create
+    assignment_form_assignment_staggered_deadline?
 
-    # Check if name and url in database is empty before webpage displays
     @due_date_all.each do |dd|
-      @due_date_nameurl_notempty = is_due_date_nameurl_notempty(dd)
-      @due_date_nameurl_notempty_checkbox = @due_date_nameurl_notempty
-      @metareview_allowed = is_meta_review_allowed?(dd)
-      @drop_topic_allowed = is_drop_topic_allowed?(dd)
-      @signup_allowed = is_signup_allowed?(dd)
-      @team_formation_allowed = is_team_formation_allowed?(dd)
-
-      if dd.due_at.present?
-        dd.due_at = dd.due_at.to_s.in_time_zone(current_user.timezonepref)
-      end
-      if  @due_date_nameurl_notempty && @due_date_nameurl_notempty_checkbox &&
-          (@metareview_allowed || @drop_topic_allowed || @signup_allowed || @team_formation_allowed)
-        break
-      end
-    end
+      due_date_nameurl_notempty_check(dd)
 
-    @assignment_questionnaires.each do |aq|
-      unless aq.used_in_round.nil?
-        @reviewvarycheck = 1
-        break
-      end
+      due_date_present(dd)
+
+      break if due_date_validation
     end
+
+    assignment_questionnaires_usage_check
+
     @due_date_all = update_nil_dd_deadline_name(@due_date_all)
     @due_date_all = update_nil_dd_description_url(@due_date_all)
 
     # only when instructor does not assign rubrics and in assignment edit page will show this error message.
-    if !empty_rubrics_list.empty? && request.original_fullpath == "/assignments/#{@assignment_form.assignment.id}/edit"
-      rubrics_needed = needed_rubrics(empty_rubrics_list)
-      flash.now[:error] = "You did not specify all the necessary rubrics. You need " + rubrics_needed +
-          " of assignment <b>#{@assignment_form.assignment.name}</b> before saving the assignment. You can assign rubrics <a id='go_to_tabs2' style='color: blue;'>here</a>."
-    end
+    rubrics_not_assigned_message
 
-    if @assignment_form.assignment.directory_path.nil? || @assignment_form.assignment.directory_path.empty?
-      flash.now[:error] = "You did not specify your submission directory."
-    end
+    assignment_directory_path_nonexist_message
   end
 
   def update
     unless params.key?(:assignment_form)
-      @assignment = Assignment.find(params[:id])
-      @assignment.course_id = params[:course_id]
-      if @assignment.save
-        flash[:note] = 'The assignment was successfully saved.'
-        redirect_to list_tree_display_index_path
-      else
-        flash[:error] = "Failed to save the assignment: #{@assignment.errors.full_messages.join(' ')}"
-        redirect_to edit_assignment_path @assignment.id
-      end
+      assignment_form_key_nonexist_case_handler
       return
     end
 
-    @assignment_form = AssignmentForm.create_form_object(params[:id])
-    @assignment_form.assignment.instructor ||= current_user
-    params[:assignment_form][:assignment_questionnaire].reject! do |q|
-      q[:questionnaire_id].empty?
-    end
+    assignment_form_retrieval
```

Comment:
- Again,  you may need to change the method name to `retrieve_assignment_form`.
- Same as two methods below, please rename your newly-added methods.